northd/ovn_northd.dl

```diff
@@ -536,31 +537,96 @@ sb.Out_Port_Group(.name = pg_name, .ports = set_empty()) :-
 
 
 /*
- * Multicast_Group: two rows per logical switch, one for flooding and one for packets with unknown
- * destinations.
+ * Multicast_Group:
+ * - three static rows per logical switch: one for flooding, one for packets
+ *   with unknown destinations, one for flooding IP multicast known traffic to
+ *   mrouters.
+ * - dynamically created rows based on IGMP groups learned by controllers.
  */
 
-function mC_FLOOD(): string   = "_MC_flood"
-function mC_UNKNOWN(): string = "_MC_unknown"
+function mC_FLOOD():         (string, integer) =
+    ("_MC_flood", 64'd32768)
+
+function mC_UNKNOWN():       (string, integer) =
+    ("_MC_unknown", 64'd32769)
+
+function mC_MROUTER_FLOOD(): (string, integer) =
+    ("_MC_mrouter_flood", 64'd32770)
+
+function mC_IP_MCAST_MIN():  (string, integer) =
+    ("_MC_ip_mcast_min", 64'd32771)
+
+function mC_IP_MCAST_MAX():  (string, integer) =
+    ("_MC_ip_mcast_max", 64'd65535)
+
 
 // TODO: check that Multicast_Group.ports should not include derived ports
 
 /* Only create flood group if the switch has enabled ports */
 sb.Out_Multicast_Group (.datapath   = datapath,
-                        .name       = mC_FLOOD(),
-                        .tunnel_key = 65535,
+                        .name       = name,
+                        .tunnel_key = tunnel_key,
                         .ports      = set_map_uuid2name(port_ids)) :-
     &SwitchPort(.lsp = lsp, .sw = &Switch{.ls = ls}),
     is_enabled(lsp.enabled),
     var datapath = uuid2name(ls._uuid),
-    var port_ids = Aggregate((datapath), group2set(lsp._uuid)).
+    (var name, var tunnel_key) = mC_FLOOD(),
+    var port_ids = Aggregate((datapath, name, tunnel_key),
```

Comment:
You don't really need to aggregate by `name` and `tunnel_key`:

```
    var port_ids = Aggregate((datapath), group2set(lsp._uuid)),
    (var name, var tunnel_key) = mC_FLOOD().
```

Reply:
> You don't really need to aggregate by `name` and `tunnel_key`:
> 
> ```
>     var port_ids = Aggregate((datapath), group2set(lsp._uuid)),
>     (var name, var tunnel_key) = mC_FLOOD().
> ```

Thanks! Done.